src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/AdyenPaymentServiceProviderPort.java

```diff
@@ -156,70 +156,54 @@ public PurchaseResult authoriseKlarnaPayment(final String merchantAccount,
         }
-        catch (ApiException ex) {
+        return handleKlarnaAuthoriseResponse(adyenCallResult, merchantAccount);
+    }
+
+    private PurchaseResult handleKlarnaAuthoriseError(AdyenCallResult<PaymentsResponse> callResult, final PaymentData paymentData) {
+        FailedCheckoutApiCall<PaymentsResponse> errorResult = (FailedCheckoutApiCall<PaymentsResponse>) callResult;
+        final String paymentTransactionId = paymentData.getPaymentTransactionExternalKey();
+
+        if (errorResult.getApiException() != null) {
+            ApiException ex = errorResult.getApiException();
             ApiError apiError = ex.getError();
-            String errorDetails = new StringBuilder()
-                    .append("status :" + apiError.getStatus())
-                    .append("errorCode :" + apiError.getErrorCode())
-                    .append("message: " + apiError.getMessage())
-                    .append("type: " +  apiError.getErrorType())
-                    .append("pspReference: " + apiError.getPspReference())
-                    .toString();
-            logger.error("completeKlarnaPayment auth error, ", errorDetails);
-            new PurchaseResult(PaymentServiceProviderResult.valueOf("error"),
-                    null,
-                    apiError.getPspReference(),
-                    apiError.getMessage(),
-                    null,
-                    paymentData.getPaymentTransactionExternalKey(),
-                    null);
-        } catch (IOException ex) {
-            new PurchaseResult(PaymentServiceProviderResult.valueOf("error"),
-                    null,
-                    null,
-                    ex.getMessage(),
-                    null,
-                    paymentData.getPaymentTransactionExternalKey(),
-                    null);
+            return new PurchaseResult(PaymentServiceProviderResult.valueOf("error"),
+                                      null,
+                                      apiError.getPspReference(),
+                                      apiError.getMessage(),
+                                      null,
+                                      paymentTransactionId,
+                                      null);
+        } else {
+            IOException ex = errorResult.getIOException();
+            return new PurchaseResult(PaymentServiceProviderResult.valueOf("error"),
+                                      null,
+                                      null,
+                                      ex.getMessage(),
+                                      null,
+                                      paymentTransactionId,
+                                      null);
         }
-
-        return purchaseResult;
     }
 
-    private PurchaseResult extractKlarnaResponse(PaymentsResponse response) {
+    private PurchaseResult handleKlarnaAuthoriseResponse(AdyenCallResult<PaymentsResponse> adyenCallResult, final String merchantAccount) {
```

Comment:
please change the name. it is confusing as it is used for both autocomplete and authorize

Reply:
Done. Check above comment.